program/src/processor.rs

```diff
@@ -22,11 +23,37 @@ use {
         instruction::AssociatedTokenAccountInstruction,
     },
     spl_token_2022_interface::{
+        check_spl_token_program_account,
         extension::{ExtensionType, StateWithExtensions},
         state::{Account, Mint},
     },
 };
 
+fn initialize_account4(
```

Comment:
i know that solana has the tradition of using `initialize_account{i}`, but i think it's poor nomenclature. could make this more verbose/better description like `initialize_account_with_close_authority`?